core/arch/arm/kernel/secure_partition.c

```diff
@@ -56,6 +56,23 @@ static void set_sp_ctx_ops(struct ts_ctx *ctx)
 	ctx->ops = _sp_ops;
 }
 
+TEE_Result sp_find_session_id(const TEE_UUID *uuid, uint32_t *session_id)
+{
+	struct sp_session *s = NULL;
+
+	TAILQ_FOREACH(s, &open_sp_sessions, link) {
+		if (!memcmp(&s->ts_sess.ctx->uuid, uuid, sizeof(*uuid))) {
+			if (s->state == sp_dead)
+				return TEE_ERROR_TARGET_DEAD;
```

Comment:
Why this special case ? The session does exist. This will cause `spmc_handle_partition_info_get()` to return FFA_ERROR/FFA_INVALID_PARAMETERS, is it the desired behavior?

Reply:
Yes, the `sp_dead` state means that the SP panicked before. The SP is in a state where it can't handle any more request. The SPMC won't handle any messages to it, so we act like it doesn't exists anymore. Currently the FF-A spec doesn't specify a way to restart a SP.
I did add the check as well to sp_get_partitions_info() to make sure we don't show SPs which panicked.

Reply:
OK